src/platforms/mesa/server/kms/display.cpp

```diff
@@ -518,13 +521,8 @@ void mgm::Display::configure_locked(
             }
             else
             {
-                glm::vec2 const logical_size{
-                    bounding_rect.size.width.as_uint32_t(),
-                    bounding_rect.size.height.as_uint32_t()};
-
-                auto const physical_size = transformation * logical_size;
```

Comment:
Whoops! This is *exactly* wrong - `logical_size = physical_size * transformation`! No wonder it blows up with non-trivial transformations.

Reply:
Well, it is accidentally right some of the time: when `transformation` is a rotation of n*90 degrees. (Which is probably the unstated assumption here.)